schedcat/locking/model.py

```diff
@@ -0,0 +1,48 @@
+import bounds
+
+class Spinlock(object):
+    def __init__(self, total_overhead=0, critical_overhead=0, apply_bounds_function=None):
+        if not apply_bounds_function:
+            apply_bounds_function = bounds.apply_pfp_lp_unordered_bounds
+        self._apply_bounds = apply_bounds_function
+        self.total_overhead = total_overhead
+        self.critical_overhead = critical_overhead
+
+    def apply_bounds(self, ts):
+        """Apply bounds and increase execution cost"""
+        self._apply_bounds(ts)
+        for _, t in enumerate(ts):
+            t.cost += t.blocked
```

Comment:
Regarding the use of the LP analysis, yes, no inflation (= tweaking of the WCET) is required.

Regarding the RTA, no! Make sure you do not confuse global and partitioned scheduling. The LP-based analysis of spin locks is for partitioned FP scheduling (where tasks never migrate). The EDF RTA that you linked to is for **global** scheduling (with migrating tasks). To analyze partitioned FP, apply the uniprocessor FP analysis (which is implemented) to each core individually.